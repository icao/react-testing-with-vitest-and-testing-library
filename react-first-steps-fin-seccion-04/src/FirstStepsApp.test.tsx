import { render, screen } from "@testing-library/react";
import { describe, expect, test, vi } from "vitest";
import { FirstStepsApp } from "./FirstStepsApp";

//! creacion de MOCK del componente ItemCounter
vi.mock("./shopping-cart/ItemCounter", () => ({
  ItemCounter: () => <div data-testid="ItemCounter">Mock ItemCounter</div>,
}));

describe("MyAwesomeApp", () => {
  test("Should match snapshot", () => {
    const { container } = render(<FirstStepsApp />);

    expect(container).toMatchSnapshot();
  });

  test("should render the correct number of ItemCounter components", () => {
    render(<FirstStepsApp />);

    const itemCounters = screen.getAllByTestId("ItemCounter").length;
    console.log("ðŸ‘‰ Numero de mocks: ", itemCounters);

    expect(itemCounters).toBe(3);
    screen.debug();
  });
});

/**
 * Dentro de nuestro componente FirtsStepsApp, tenemos un array de objetos que representan los items en nuestro carro de compras.
 * Estos objetos tienen dos propiedades: productName y quantity.
 * No podemos testear el componente con la lista que esta mapeando, ya que no sabemos cuantos items hay en el array.
 * Para este test podemos usar un componente MOCK que nos permita controlar los items que se estan renderizando. Y asi corroborar que el componente se esta renderizando correctamente la lista de items.
 */
