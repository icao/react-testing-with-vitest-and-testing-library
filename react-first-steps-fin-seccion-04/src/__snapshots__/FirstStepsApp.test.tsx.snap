// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyAwesomeApp > Should match snapshot 1`] = `
<div>
  <h1>
    Carrito de compras
  </h1>
  <div
    data-testid="ItemCounter"
  >
    Mock ItemCounter
  </div>
  <div
    data-testid="ItemCounter"
  >
    Mock ItemCounter
  </div>
  <div
    data-testid="ItemCounter"
  >
    Mock ItemCounter
  </div>
</div>
`;
